### PR TITLE
David

### PR DESCRIPTION
**Pull request by schwiti6190**

### CODE REVIEW

File: FillableFieldworkAIDriver.lua [hunk 1]
```diff
@@ -70,11 +70,11 @@ function FillableFieldworkAIDriver:driveUnloadOrRefill()
 		self:clearInfoText('NO_SELECTED_FILLTYPE')
 	end
 	local isNearWaitPoint, waitPointIx = self.course:hasWaitPointWithinDistance(self.ppc:getCurrentWaypointIx(), 10)
-	self.waitPointIx = waitPointIx
+	local isNearPastWaitPoint, pastWaitPointIX = (self.ppc:getCurrentWaypointIx()-5)>1 and self.course:hasWaitPointWithinDistance(self.ppc:getCurrentWaypointIx()-5, 10)
```
Comment (pvaiko) [inline FillableFieldworkAIDriver.lua:73]: why -5? why not just based on distance? 

note there's a PurePursuitController:getRelevantWaypointIx() which is the waypoint closest to us

Comment (schwiti6190) [reply FillableFieldworkAIDriver.lua:73]: @pvaiko I would like to have self.course:hasWaitPointWithinDistance(self.ppc:getCurrentWaypointIx(), 10), but _backwards_ so I can check once we've left the waypoint behind. Not sure about this one "getRelevantWaypointIx" as I would have to maybe check more waypoints before it.

Comment (pvaiko) [reply FillableFieldworkAIDriver.lua:73]: but hasWaitpintWithinDistance searches backwards and forward too, in both directions.

Comment (pvaiko) [reply FillableFieldworkAIDriver.lua:73]: your problem may be that the _current_ waypoint is always in front of the vehicle, that's why I suggested to use the relevant waypoint or last passed waypoint.

Comment (schwiti6190) [reply FillableFieldworkAIDriver.lua]: @pvaiko  now I get it, so self.course:hasWaitPointWithinDistance(self.ppc:getRelevantWaypointIx(), 10), would basicly check 10m ahead and behind of roughly the current vehicle position ?

Comment (pvaiko) [reply FillableFieldworkAIDriver.lua]: yes

**Pull request merged**